packages/ui/node/index.ts

```diff
@@ -1,13 +1,25 @@
 import { fileURLToPath } from 'url'
-import { resolve } from 'pathe'
+import { basename, resolve } from 'pathe'
 import sirv from 'sirv'
 import type { Plugin } from 'vite'
 
-export default (base = '/__vitest__/') => {
+export default (base = '/__vitest__/', resolveCoverageFolder?: () => string | undefined) => {
+  let coveragePath: string | undefined
+  let coverageFolder: string | undefined
   return <Plugin>{
     name: 'vitest:ui',
     apply: 'serve',
+    configResolved() {
+      coverageFolder = resolveCoverageFolder?.() ?? undefined
+      coveragePath = coverageFolder ? `/${basename(coverageFolder)}/` : undefined
+      if (coveragePath && base === coveragePath)
+        throw new Error(`The base path and the coverage path cannot be the same: ${base}`)
+    },
     async configureServer(server) {
+      coverageFolder && server.middlewares.use(coveragePath!, sirv(coverageFolder, {
```

Comment:
Can't you just get all the fields that you use in`resolveCoverageFolder`  from `server.config.test`?

Reply:
done